src/main/java/dog/pawbook/logic/commands/EditDogCommand.java

```diff
@@ -52,36 +52,6 @@ public EditDogCommand(Integer id, EditDogDescriptor editDogDescriptor) {
         super(id, editDogDescriptor);
     }
 
-    @Override
-    public CommandResult execute(Model model) throws CommandException {
-        requireNonNull(model);
-
-        if (!model.hasEntity(id)) {
-            throw new CommandException(Messages.MESSAGE_INVALID_DOG_ID);
-        }
-        Entity targetEntity = model.getEntity(id);
-
-        if (!(targetEntity instanceof Dog)) {
-            throw new CommandException(Messages.MESSAGE_INVALID_DOG_ID);
-        }
-        Dog targetDog = (Dog) targetEntity;
-        Dog editedDog = createEditedEntity(targetEntity, editEntityDescriptor);
-
-        if (!targetDog.equals(editedDog) && model.hasEntity(editedDog)) {
-            throw new CommandException(getDuplicateEntityMessage());
-        }
-
-        // special handling of Owner ID change
-        int originalOwnerId = targetDog.getOwnerId();
-        int editedOwnerId = editedDog.getOwnerId();
-        if (originalOwnerId != editedOwnerId) {
-            changeOwner(model, originalOwnerId, editedOwnerId);
-        }
-
-        model.setEntity(id, editedDog);
-        return new CommandResult(getSuccessMessage(editedDog));
-    }
```

Comment:
Do you mind explaining why this is taken away? I cannot seem to find a reason that the override should not happen.

Reply:
I think this part is causing some bugs. Coz from there's no execute() in editOwnerCommand and editProgramCommand. So this execute() should inherit from editDogCommand as well since it already inherited the editEntityCommand.

Reply:
Restore it back, it is essential, or else when the owner ID of a dog changes, the old owner will still have the dog

Reply:
Done!